### PR TITLE
support international keyboards

### PR DESCRIPTION
**Pull request by p3r7**
`keyboard.code` returns "low-level" alias for HID codes.

`keyboard.char` produces international keyboard remapped chars.
it gets called just after `keyboard.code`.

### CODE REVIEW

**Comment by p3r7** [comment]
also, without using those callbacks, any script can check if a modifier key is pressed at any given time using `keyboard.state.<HID_CODE_ALIAS>`.

for example:
```lua
function redraw()
  -- [...]
  screen.level(keyboard.state.ESC and 15 or 0)
  -- [...]
end
```

in fact any key can be treated as a modifier this way (e.g. `keyboard.state.A`, `keyboard.state['0']`...) but for alphanumeric characters this won't honor international keyboard layouts.

File: KEY.lua [hunk 1]
```diff
@@ -18,8 +18,15 @@ function init()
 end
 
 function keyboard.code(code, value)
-  down = value == 0
-  buffer = code
+  down = value == 1
```
Comment (p3r7) [inline KEY.lua]: key down (pressed) is when value is 1

File: KEY.lua [hunk 2]
```diff
@@ -18,8 +18,15 @@ function init()
 end
 
 function keyboard.code(code, value)
-  down = value == 0
-  buffer = code
+  down = value > 0
```
Comment (p3r7) [inline KEY.lua:21]: key down (pressed) is when value is 1 (or in some rare instances 2)

Comment (tyleretters) [reply KEY.lua:21]: !

**Comment by tyleretters** [comment]
@p3r7 thank you!!

**Pull request merged**